recipes/paho-mqtt-c/all/conanfile.py

```diff
@@ -1,27 +1,34 @@
-import os
 from conans import CMake, ConanFile, tools
 from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.29.1"
+
 
 class PahoMqttcConan(ConanFile):
     name = "paho-mqtt-c"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/eclipse/paho.mqtt.c"
-    topics = ("MQTT", "IoT", "eclipse", "SSL", "paho", "C")
+    topics = ("mqtt", "iot", "eclipse", "ssl", "tls", "paho", "c")
     license = "EPL-2.0"
     description = "Eclipse Paho MQTT C client library for Linux, Windows and MacOS"
     exports_sources = ["CMakeLists.txt", "patches/*"]
     generators = "cmake"
     settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False],
-               "fPIC": [True, False],
-               "ssl": [True, False],
-               "samples": [True, False],
-               "asynchronous": [True, False]}
-    default_options = {"shared": False,
-                       "fPIC": True,
-                       "ssl": True,
-                       "asynchronous" : True,
-                       "samples": False}
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "ssl": [True, False],
+        "asynchronous": [True, False],
+        "samples": [True, False, "deprecated"]
```

Comment:
don't forget to remove `self.info.options.samples` in `package_id()`, and display a warning in `configure()` if `self.options.samples != "deprecated"`